docker/docker_image_dest.go

```diff
@@ -120,15 +145,26 @@ func (d *dockerImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobI
 			return types.BlobInfo{Digest: inputInfo.Digest, Size: size}, nil
 		}
 	}
+	mountQuery := prepareMountQuery(d.mount, inputInfo.Digest.String())
 
 	// FIXME? Chunked upload, progress reporting, etc.
 	uploadPath := fmt.Sprintf(blobUploadPath, reference.Path(d.ref.ref))
+	if mountQuery != "" {
+		logrus.Debugf("Trying mount: %s", mountQuery)
+		uploadPath = fmt.Sprintf("%s?%s", uploadPath, mountQuery)
+	}
 	logrus.Debugf("Uploading %s", uploadPath)
 	res, err := d.c.makeRequest(context.TODO(), "POST", uploadPath, nil, nil)
 	if err != nil {
 		return types.BlobInfo{}, err
 	}
 	defer res.Body.Close()
+	if res.StatusCode == http.StatusCreated {
+		size := getBlobSize(res)
```

Comment:
https://github.com/docker/distribution/blob/master/docs/spec/api.md#mount-blob explicitly says that `Content-Length` of the response must be zero, so I don’t think this will work. `d.HasBlob` will probably be necessary to get the size.